bin/start_lke.sh

```diff
@@ -0,0 +1,348 @@
+  echo >&2 "Python 3 must be installed to continue"
+  exit 1
+fi
+
+if ! command -v node >/dev/null; then
+  if [ -x "${script_dir}/../pulumi/python/venv/bin/pulumi" ]; then
+    echo "Adding to [${script_dir}/../pulumi/python/venv/bin] to PATH"
+    export PATH="${script_dir}/../pulumi/python/venv/bin:$PATH"
+
+    if ! command -v node >/dev/null; then
+      echo >&2 "NodeJS must be installed to continue"
+      exit 1
+    fi
+  else
+    echo >&2 "NodeJS must be installed to continue"
+    exit 1
+  fi
+fi
+
+if ! command -v git >/dev/null; then
+  echo >&2 "git must be installed to continue"
+  exit 1
+fi
+
+if ! command -v make >/dev/null; then
+  echo >&2 "make is not installed - it must be installed if you intend to build NGINX Kubernetes Ingress Controller from source."
+fi
+
+if ! command -v docker >/dev/null; then
+  echo >&2 "docker is not installed - it must be installed if you intend to build NGINX Kubernetes Ingress Controller from source."
+fi
+
+# Check to see if the user is logged into Pulumi
+if ! pulumi whoami --non-interactive >/dev/null 2>&1; then
+  pulumi login
+
+  if ! pulumi whoami --non-interactive >/dev/null 2>&1; then
+    echo >&2 "Unable to login to Pulumi - exiting"
+    exit 2
+  fi
+fi
+
+if [ ! -f "${script_dir}/../config/pulumi/environment" ]; then
+  touch "${script_dir}/../config/pulumi/environment"
+fi
+
+if ! grep --quiet '^PULUMI_STACK=.*' "${script_dir}/../config/pulumi/environment"; then
+  read -r -e -p "Enter the name of the Pulumi stack to use in all projects: " PULUMI_STACK
+  echo "PULUMI_STACK=${PULUMI_STACK}" >>"${script_dir}/../config/pulumi/environment"
+fi
+
+# Do we have the submodule source....
+#
+# Note: We had been checking for .git, but this is not guaranteed to be
+# there if we build the docker image or use a tarball. So now we look
+# for the src subdirectory which should always be there.
+#
+if [[ -d "${script_dir}/../pulumi/python/kubernetes/applications/sirius/src/src" ]]; then
+  echo "Submodule source found"
+else
+  # Error out with instructions.
+  echo "Bank of Sirius submodule not found"
+  echo " "
+  echo "Please run:"
+  echo "    git submodule update --init --recursive --remote"
+  echo "Inside your git directory and re-run this script"
+  echo ""
+  echo >&2 "Unable to find submodule - exiting"
+  exit 3
+fi
+
+source "${script_dir}/../config/pulumi/environment"
+echo "Configuring all Pulumi projects to use the stack: ${PULUMI_STACK}"
+
+# Create the stack if it does not already exist
+# We skip over the tools directory, because that uses a unique stack for setup of the
+# kubernetes components for installations without them.
+find "${script_dir}/../pulumi/python" -mindepth 1 -maxdepth 7 -type f -name Pulumi.yaml -not -path "*/tools/*" -execdir pulumi stack select --create "${PULUMI_STACK}" \;
+
+if [[ -z "${LINODE_TOKEN+x}" ]]; then
+  echo "LINODE_TOKEN not set"
+  if ! grep --quiet '^LINODE_TOKEN=.*' "${script_dir}/../config/pulumi/environment"; then
```

Comment:
Make sure you check for GNU grep OR that this grep command also works in the MacOS grep flavor.

Reply:
This works on either; but this is also a really great point for multi-arch type installs like this. Adding this as a checklist item.